docs/client-specifications/configuration.md

```diff
@@ -0,0 +1,147 @@
+
+The `SimpleCacheClient` constructor should accept a required argument whose type is our `Configuration` interface.  e.g.:
+
+```typescript
+import * as config from '@gomomento/sdk/config';
+
+const client = SimpleCacheClient({config: config.Dev.latest()});
+```
+
+# Pre-built Configuration Objects
+
+## Requirements for each pre-built configuration objects
+
+Each SDK must provide a namespace (via the idiomatic construct in the relevant programming language) that contains
+an enumerated list of pre-built configuration objects.
+
+All configuration objects must provide a `latest()` function that returns the latest "blessed" defaults from Momento.
+Documentation must clearly indicate that these values may change over time if Momento finds in the future that some
+different default values will provide a better or more performant user experience.
+
+All configuration objects must also provide a `default()` function that returns the original "blessed" defaults from
+Momento.  Documentation must clearly state that these values will not change over time; users can rely on them to
+be stable across new minor version releases of the SDKs.
+
+The `SimpleCacheClient` constructor should expose a required argument that is an instance of
+
+## Required pre-built configuration objects
+
+Each SDK must provide at least the following out-of-the-box configuration objects:
+
+```typescript
+import * as config from '@gomomento/sdk/config';
+
+/**
+ * Dev provides defaults suitable for a medium-to-high-latency dev environment.  Permissive timeouts, retries, potentially
+ * a higher number of connections, etc.
+ */
+const devConfig = config.Dev.latest();
+
+/**
+ * ProdLowLatency provides defaults suitable for a low-latency prod environment.  This is for the use case where the
+ * client is running on an application server in the same region as the Momento service.  It has more agressive timeouts
+ * and retry behavior, to ensure that cache unavailability doesn't force unacceptably high application latencies.
+ */
+const prodConfig = config.ProdLowLatency.latest();
+```
+
+## Configuration interface
+
+The configuration interface provides nested objects for retry strategy, middleware, and transport:
+
+```typescript
+interface Configuration {
+  retryStrategy: RetryStrategy,
+  middleware: Middleware,
+  transportStrategy: TransportStrategy
+}
+```
+
+### RetryStrategy
+
+The `RetryStrategy` provides a function that is called when a request fails.  This function is responsible for determining
+whether the request should be retried, and if so, how long the client should delay before retrying:
+
+```typescript
+interface RetryStrategy {
+  /**
+   * Returns the number of milliseconds after which the request should be retried, or `undefined` if the request should
+   * not be retried.  `0` means retry immediately.
+   */
+  determineWhenToRetryRequest(grpcResult: GrpcResult, grpcRequest: GrpcRequest, attemptNumber: number): Promise<number | undefined>;
```

Comment:
If we were to support http as transport down line would we just add another field on this interface with different signature like this? 
```typescript
   determineWhenToRetryRequest(httpResult: httpResult, HttpRequest: HttpRequest, attemptNumber: number): Promise<number | undefined>;
```

Just want to make sure im understanding how your thinking about modifying this down line with different transport strategies. Just warning flag went off when I see GRPC specific things in method signature exposed to user.

Reply:
Yeah, again - this would never be visible to a user that chose to use one of our out-of-the-box configurations.  But it would be visible if they wanted to implement a custom configuration.

If they just want to tune a "max retries" setting or something, we may have a builder like:

```typescript
const retryStrategy = new BasicRetryStrategy({
   maxAttempts: 5,
   exponentialBackOffFactor: 2,
   jitterNanos: 500
});
```

That thing would implement the interface, and then they'd plug that in to their Configuration constructor.  So they will only ever see the gRPC stuff in the signature if they are doing a custom implementation of a RetryStrategy, which is most likely an indicator that we should have provided more/better out-of-the-box ones.

But yes, I also think/hope that the interface could be extended to support alternate signatures for HTTP in the future.

Thanks for reviewing this thoroughly and challenging.  I'll add more example code to cover this case as well, but please continue to raise questions or push back if you still have concerns!

Reply:
added some examples in https://github.com/momentohq/standards-and-practices/pull/14/commits/7c54ee1016b39bef24280ead497e121de398e974